pkg/kubeapiserver/resource_handler.go

```diff
@@ -49,6 +49,29 @@ func (r *ResourceHandler) ServeHTTP(w http.ResponseWriter, req *http.Request) {
 	gvr := schema.GroupVersionResource{Group: requestInfo.APIGroup, Version: requestInfo.APIVersion, Resource: requestInfo.Resource}
 
 	clusterName := request.ClusterNameValue(req.Context())
+
+	var (
+		cluster *clusterv1alpha2.PediaCluster
+		err     error
+	)
+	// When clusterName not empty, first check cluster whether exist
+	if clusterName != "" {
+		if cluster, err = r.clusterLister.Get(clusterName); err != nil {
+			if !apierrors.IsNotFound(err) {
+				klog.ErrorS(err, "Failed to handle resource request, not get cluster from cache", "cluster", clusterName, "resource", gvr)
+				responsewriters.ErrorNegotiated(
+					apierrors.NewInternalError(err),
+					Codecs, gvr.GroupVersion(), w, req,
+				)
+				return
+			}
+			responsewriters.ErrorNegotiated(
+				apierrors.NewBadRequest(err.Error()),
```

Comment:
For errors that have been identified, it is better to use a fixed format or string, for example `not found requested cluster`